gateway/internal/middleware/authorization.go

```diff
@@ -0,0 +1,70 @@
+package middleware
+
+import (
+	"context"
+	"github.com/catness812/e-petitions-project/gateway/internal/config"
+	"github.com/catness812/e-petitions-project/gateway/internal/user/pb"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type AuthMiddleware struct {
+	userClient pb.UserControllerClient
+	cfg        *config.PermissionsConfig
+}
+
+func NewAuthorizationMiddleware(userClient pb.UserControllerClient, rbacCfg *config.PermissionsConfig) *AuthMiddleware {
+	return &AuthMiddleware{userClient: userClient, cfg: rbacCfg}
+}
+
+func (auth *AuthMiddleware) Authorize(action, resourceCode string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		mail, ok := c.Get("userEmail")
+		if !ok {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{
+				"error":   "Unauthorized",
+				"message": "You do not have permission to perform this action.",
```

Comment:
repetitive error I mean

Reply:
What do you mean repetetive error? If there is no email in the context why should i let the request through?